src/AcceleratedDubins.jl

```diff
@@ -501,63 +449,43 @@ Sample single path for plotting
-    elseif Integer(path.type) == Integer(RSR)
-        ang1 = path.origin[3]+pi/2
-        ang2 = ang1 - path.lengths[1]/path.r[1]
-        confx, confy = sample_arc(path.origin, ang1, ang2, path.r[1], resolution, 1)
-        A = [confx[end], confy[end]]
-        B = point_by_angle(A, ang2-pi/2, path.lengths[2])
-        push!(confx, B[1])
-        push!(confy, B[2])
-        ang3 = ang2 - path.lengths[3]/path.r[3]
-        confx2, confy2 = sample_arc(B, ang2, ang3, path.r[3], resolution, 1)
-        append!(confx, confx2)
-        append!(confy, confy2)
-        return confx, confy
-    end
-    if Integer(path.type) == Integer(LSL)
-        ang1 = path.origin[3]-pi/2
-        ang2 = ang1 + path.lengths[1]/path.r[1]
-        confx, confy = sample_arc(path.origin, ang1, ang2, path.r[1], resolution, -1)
-        A = [confx[end], confy[end]]
-        B = point_by_angle(A, (ang2+pi/2), path.lengths[2])
-        push!(confx, B[1])
-        push!(confy, B[2])
-        ang3 = ang2 + path.lengths[3]/path.r[3]
-        confx2, confy2 = sample_arc(B, ang2, ang3, path.r[3], resolution, -1)
-        append!(confx, confx2)
-        append!(confy, confy2)
-        return confx, confy
-    end
-    if Integer(path.type) == Integer(LSR)
-        ang1 = path.origin[3]-pi/2
-        ang2 = ang1 + path.lengths[1]/path.r[1]
-        confx, confy = sample_arc(path.origin, ang1, ang2, path.r[1], resolution, -1)
-        A = [confx[end], confy[end]]
-        B = point_by_angle(A, ang2+(pi/2), path.lengths[2])
-        push!(confx, B[1])
-        push!(confy, B[2])
-        ang3 = ang2 - pi - path.lengths[3]/path.r[3]
-        confx2, confy2 = sample_arc(B, mod2pi(ang2+pi), ang3, path.r[3], resolution, 1)
-        append!(confx, confx2)
-        append!(confy, confy2)
-        return confx, confy
-    end
-    if Integer(path.type) == Integer(RSL)
-        ang1 = path.origin[3]+pi/2
-        ang2 = ang1 - path.lengths[1]/path.r[1]
-        confx, confy = sample_arc(path.origin, ang1, ang2, path.r[1], resolution, 1)
-        A = [confx[end], confy[end]]
-        B = point_by_angle(A, ang2-(pi/2), path.lengths[2])
-        push!(confx, B[1])
-        push!(confy, B[2])
-        ang3 = (ang2 + pi) + path.lengths[3]/path.r[3]
-        confx2, confy2 = sample_arc(B, mod2pi(ang2+pi), ang3, path.r[3], resolution, -1)
-        append!(confx, confx2)
-        append!(confy, confy2)
+    else
+        confx::Vector{Float64} = []
+        confy::Vector{Float64} = []
+
+        # Sample first arc
+        for l in 0.:resolution:path.lengths[1]
+            x, y, _ = get_configuration(path, l)
+            push!(confx, x)
+            push!(confy, y)
+        end
+
+        # Add last point of first arch
+        x, y, theta = get_configuration(path, path.lengths[1])
+        push!(confx, x)
+        push!(confy, y)
+
+        # Sample line segment
+        x, y = point_by_angle([x, y], theta, path.lengths[2])       
+        push!(confx, x)
+        push!(confy, y)
```

Comment:
As originally, straight segment is sampled as two points. If this is not desirable anymore, whole function could be reduced to a single for-loop, similarly to first arc sampling